src/Bicep.Core/Syntax/ProviderDeclarationSyntax.cs

```diff
@@ -41,14 +42,14 @@ public ProviderDeclarationSyntax(IEnumerable<SyntaxBase> leadingNodes, Token key
 
         public IdentifierSyntax? Alias => (this.AsClause as AliasAsClauseSyntax)?.Alias;
 
-        private StringSyntax ProviderPath => SyntaxFactory.CreateStringLiteral($@"{OciArtifactReferenceFacts.Scheme}:{LanguageConstants.BicepPublicMcrRegistry}/bicep/providers/{this.Specification.Name}:{this.Specification.Version}");
-
         public override TextSpan Span => TextSpan.Between(this.Keyword, TextSpan.LastNonNull(this.SpecificationString, this.WithClause, this.AsClause));
 
         SyntaxBase IForeignArtifactReference.ReferenceSourceSyntax => ProviderPath;
 
         public override void Accept(ISyntaxVisitor visitor) => visitor.VisitProviderDeclarationSyntax(this);
 
+        private StringSyntax ProviderPath => SyntaxFactory.CreateStringLiteral($@"{OciArtifactReferenceFacts.Scheme}:{FeatureProvider.ReadEnvVar("BICEP_REGISTRY_FQDN", LanguageConstants.BicepPublicMcrRegistry)}/bicep/providers/{this.Specification.Name}:{this.Specification.Version}");
```

Comment:
I'm inclined to agree with Ryan here -- this feels extremely temporary. In the long term, we definitely want to support multiple registries, and that configuration will need to come from `bicepconfig.json` instead of the environment. 

Since we know that this is just temporary scaffolding for an experimental feature, can we give the environment variable a less official sounding name? Maybe `__EXPERIMENTAL_BICEP_REGISTRY_FQDN` or `BICEP_REGISTRY_FQDN__EXPERIMENTAL`?